src/rules/no-store-async.ts

```diff
@@ -0,0 +1,44 @@
+import { createRule } from "../utils"
+import type * as ESTree from "estree"
+
+export default createRule("no-store-async", {
+  meta: {
+    docs: {
+      description:
+        "disallow using async/await inside svelte stores because it causes issues with the auto-unsubscribing features",
+      category: "Possible Errors",
+      recommended: true,
+      default: "error",
+    },
+    schema: [],
+    messages: {
+      unexpected: "Do not pass async functions to svelte stores.",
+    },
+    type: "problem",
+  },
+  create(context) {
+    return {
+      CallExpression(node: ESTree.CallExpression) {
+        if (node.callee.type !== "Identifier") return
+        const { name } = node.callee
+        if (name !== "writable" && name !== "readable" && name !== "derived")
+          return
+        const [, fn] = node.arguments
+        if (fn.type !== "ArrowFunctionExpression" || !fn.async) return
```

Comment:
I think we should check if the argument exists.

```suggestion
        if (!fn || fn.type !== "ArrowFunctionExpression" || !fn.async) return
```

Reply:
fixed! Sorry for that!

https://github.com/ota-meshi/eslint-plugin-svelte/pull/225/commits/99e0996c1866e285e6d3d4d92172f037415e3500